common/desktop-exports

```diff
@@ -30,14 +30,24 @@ export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$LIBGL_DRIVERS_PATH
 export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/var/lib/snapd/lib/gl
 
 # Mir
-# Put mir-libs first because it's easy to end up with a snap-internal copy
-export LD_LIBRARY_PATH=$SNAP/mir-libs/$ARCH:$LD_LIBRARY_PATH
-export MIR_CLIENT_PLATFORM_PATH=$SNAP/mir-libs/$ARCH/mir/client-platform
-export MIR_SOCKET=/run/user/$(id -u)/mir_socket
-if [ ! -S "$MIR_SOCKET" -a -S /run/mir_socket ]; then
+if [ -S /run/user/$(id -u)/mir_socket ]; then
+  export MIR_SOCKET=/run/user/$(id -u)/mir_socket
+elif [ -S /run/mir_socket ]; then
   # Fall back to global location if there is no user server (e.g. on a kiosk)
   export MIR_SOCKET=/run/mir_socket
 fi
+if [ -d $SNAP/mir-libs/$ARCH ]; then
+  export MIR_CLIENT_PLATFORM_PATH=$SNAP/mir-libs/$ARCH/mir/client-platform
+  # Put mir-libs first because it's easy to end up with a snap-internal copy
+  export LD_LIBRARY_PATH=$SNAP/mir-libs/$ARCH:$LD_LIBRARY_PATH
+elif [ -d $RUNTIME/usr/lib/$ARCH/mir ]; then
+  # As of this writing, ubuntu-app-platform holds a copy of the Mir libraries,
+  # but may remove them in favor of mir-libs.  Use them if present.
+  export MIR_CLIENT_PLATFORM_PATH=$RUNTIME/usr/lib/$ARCH/mir/client-platform
```

Comment:
Do you mind adding a comment for that? I'm afraid in future cleanups swiping it.